rpcserver.go

```diff
@@ -637,6 +637,73 @@ func createVinList(mtx *wire.MsgTx) []btcjson.Vin {
 	return vinList
 }
 
+// createVinList returns a slice of JSON objects for the inputs of the passed
+// transaction.
+func createVinListPrevOut(s *rpcServer, mtx *wire.MsgTx, chainParams *chaincfg.Params, vinExtra int) []btcjson.VinPrevOut {
+	vinList := make([]btcjson.VinPrevOut, len(mtx.TxIn))
+
+	// we don't call fetchInputTransactions() here because it may not be
+	// necessary so we defer until it is.  ( lazy load )
+	var txStore *blockchain.TxStore
+
+	for i, v := range mtx.TxIn {
+		if blockchain.IsCoinBaseTx(mtx) {
+			vinList[i].Coinbase = hex.EncodeToString(v.SignatureScript)
+		} else {
+			vinList[i].Txid = v.PreviousOutPoint.Hash.String()
+			vinList[i].Vout = v.PreviousOutPoint.Index
+
+			// The disassembled string will contain [error] inline
+			// if the script doesn't fully parse, so ignore the
+			// error here.
+			disbuf, _ := txscript.DisasmString(v.SignatureScript)
+			vinList[i].ScriptSig = new(btcjson.ScriptSig)
+			vinList[i].ScriptSig.Asm = disbuf
+			vinList[i].ScriptSig.Hex = hex.EncodeToString(v.SignatureScript)
+
+			// If vinExtra flag is set then we grab extra data from the
+			// previous transaction output.
+			if vinExtra == 1 {
```

Comment:
With the other suggestion, this would just then be:

```Go
if vinExtra == 1 && len(txStore) > 0 {
	txData := txStore[v.PreviousOutPoint.Hash]
	if txData != nil {
		...
	}
}